packages/lwc/index.d.ts

```diff
@@ -207,61 +207,58 @@ declare module 'lwc' {
      * @param config configuration object for the adapter
      */
     export function wire<
-        Config extends StringKeyedRecord = any,
-        Value = any,
-        Context extends StringKeyedRecord = any
+        Config extends StringKeyedRecord,
+        Value,
+        Context extends StringKeyedRecord
     >(
         adapter:
             | WireAdapterConstructor<Config, Value, Context>
             | LegacyWireAdapterConstructor<Config, Value>,
         config?: WireConfigValue<Config>
     ): PropertyDecorator;
 
-    type LegacyWireAdapterConstructor<Config = any, Value = any> = (config?: Config) => Value;
+    type LegacyWireAdapterConstructor<Config, Value> = (config?: Config) => Value;
     type WireConfigValue<Config extends StringKeyedRecord = StringKeyedRecord> = {
         // wire reactive variables are strings prefixed with '$' so the config value can just be string
         [K in keyof Config]: Config[K] | string;
     };
     type ContextValue<Context extends StringKeyedRecord = StringKeyedRecord> = Context;
 
-    interface WireAdapter<
-        Config extends StringKeyedRecord = any,
-        Context extends StringKeyedRecord = any
-    > {
+    interface WireAdapter<Config extends StringKeyedRecord, Context extends StringKeyedRecord> {
         update(config: WireConfigValue<Config>, context?: ContextValue<Context>): void;
         connect(): void;
         disconnect(): void;
     }
 
-    type WireDataCallback<Value = any> = (value: Value) => void;
+    type WireDataCallback<Value> = (value: Value) => void;
     type WireAdapterSchemaValue = 'optional' | 'required';
 
-    interface ContextConsumer<Context extends StringKeyedRecord = any> {
+    interface ContextConsumer<Context extends StringKeyedRecord> {
         provide(newContext: ContextValue<Context>): void;
     }
 
-    interface ContextProviderOptions<Context extends StringKeyedRecord = any> {
+    interface ContextProviderOptions<Context extends StringKeyedRecord> {
         consumerConnectedCallback: (consumer: ContextConsumer<Context>) => void;
         consumerDisconnectedCallback?: (consumer: ContextConsumer<Context>) => void;
     }
 
     interface WireAdapterConstructor<
-        Config extends StringKeyedRecord = any,
-        Value = any,
-        Context extends StringKeyedRecord = any
+        Config extends StringKeyedRecord,
+        Value,
+        Context extends StringKeyedRecord
```

Comment:
`Context` is optional and for other places of occurrences too. I think even if it is not exported, it can still be used so this should have a default type.

Reply:
It's optional but this type should technically only be used internally (even though it's public), and the two places it's used provide default values for it. Let's go with this for now and we can loosen it up later if necessary.